### PR TITLE
ENH: add windows support

### PR DESCRIPTION
**Pull request by RubyLiu206**
Windows support was added mostly by:

- Fixing path separator issues
- Making sure test executables existed within a regular windows install
- Adding support python programs if this wasn't the case.

Please fill in the fields below to submit a pull request.  The more information
that is provided, the better.

**Fixes issue #**:

**Description of the changes being introduced by the pull request**:

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: in_toto/models/link.py
Comment (alyptik) [inline in_toto/models/link.py]: Small spelling error; probably meant to be "features."

File: in_toto/models/metadata.py
Comment (alyptik) [inline in_toto/models/metadata.py]: Same spellng error.

File: in_toto/models/metadata.py
Comment (alyptik) [inline in_toto/models/metadata.py]: Same spellng error.

File: in_toto/models/metadata.py
Comment (alyptik) [inline in_toto/models/metadata.py]: Same spellng error.

File: in_toto/models/metadata.py
Comment (alyptik) [inline in_toto/models/metadata.py]: Same spellng error.

File: in_toto/verifylib.py
Comment (alyptik) [inline in_toto/verifylib.py]: Is there a reason for the extra comma here?

File: tests/test_in_toto_verify.py
Comment (alyptik) [inline tests/test_in_toto_verify.py:87]: Was the indentation level change here intended?

**Review by alyptik — CHANGES_REQUESTED** [review]
Aside from a few spelling errors and small nitpicks, looks good to me.

File: in_toto/models/metadata.py
Comment (SantiagoTorres) [reply in_toto/models/metadata.py]: I'd probably drop these changes from this PR

File: in_toto/verifylib.py
Comment (SantiagoTorres) [reply in_toto/verifylib.py]: os.path.join needs two params, but passing an empty string doesn't work like it should in certain python versions...

File: tests/test_in_toto_verify.py
Comment (SantiagoTorres) [reply tests/test_in_toto_verify.py:87]: is this na indentation change or just a *new* copytree statement. I think this is right

File: in_toto/verifylib.py
Comment (alyptik) [reply in_toto/verifylib.py]: Ah, that's a strange quirk, noted.

File: tests/test_in_toto_verify.py
Comment (alyptik) [reply tests/test_in_toto_verify.py:87]: Yeah I figured it is possibly intended, but I don't have enough context to be sure, so just figured I'd mention it.

**Comment by trishankatdatadog** [comment]
@SantiagoTorres @alyptik How can we help push these changes ASAP?

We are happy test and address the comments in this PR. @ofek

**Comment by trishankatdatadog** [comment]
I think we should add AppVeyor to make sure that in-toto works on Windows. We don't have to do it right now with this PR, but we should add testing against Windows.

File: in_toto/models/link.py
Comment (trishankatdatadog) [inline in_toto/models/link.py]: What is there TODO here?

File: in_toto/models/metadata.py
Comment (trishankatdatadog) [reply in_toto/models/metadata.py]: What are these missing features?

File: in_toto/models/metadata.py
Comment (trishankatdatadog) [reply in_toto/models/metadata.py]: What are these missing features?

File: in_toto/models/metadata.py
Comment (trishankatdatadog) [reply in_toto/models/metadata.py]: What are these missing features?

File: in_toto/models/metadata.py
Comment (trishankatdatadog) [reply in_toto/models/metadata.py]: What are these missing features?

File: in_toto/runlib.py
Comment (trishankatdatadog) [inline in_toto/runlib.py]: This needs to be removed.

File: in_toto/runlib.py
Comment (trishankatdatadog) [inline in_toto/runlib.py]: This needs to be removed.

File: tests/common.py
Comment (trishankatdatadog) [inline tests/common.py]: This needs to be removed.

File: tests/common.py
Comment (trishankatdatadog) [inline tests/common.py]: This needs to be removed.

File: in_toto/runlib.py
Comment (trishankatdatadog) [inline in_toto/runlib.py]: I am not convinced that this PR would fix all issues on Windows.

For example, are there checks to prevent absolute paths such as `C://Program Files/in-toto/run.py`?

**Review by trishankatdatadog — CHANGES_REQUESTED** [review]

File: in_toto/models/metadata.py
Comment (SantiagoTorres) [reply in_toto/models/metadata.py]: This is part of another pr, #215 . I think these markers were put in there when we were discussing what features to add..

File: in_toto/models/metadata.py
Comment (SantiagoTorres) [reply in_toto/models/metadata.py]: For this and all of the above. We need to remove these comments as they are not relevant to this changeset.

File: in_toto/models/link.py
Comment (SantiagoTorres) [reply in_toto/models/link.py]: Same as above...

**Comment by SantiagoTorres** [comment]
@RubyLiu206 there are a lot of stray comments from our discussion of PR #215 

@trishankatdatadog the appveyor aspect was added on #207, but we haven't worked on it until this PR is merged.

**Comment by RubyLiu206** [comment]
I will delete those today.

**Comment by trishankatdatadog** [comment]
Okay, thanks, everyone 🙇 

We'd like to get this package, please, before releasing a new version of tool (cc @ofek)

**Comment by trishankatdatadog** [comment]
I just realized looking at an [example of a link metadata file produced on Windows](https://github.com/DataDog/integrations-core/blob/16e5789d22b742f49d391ead2a4886d69ca89e83/.links/tag.c295cf63.link) that `in-toto` should accept Windows path separators, even if it was somehow written and signed.

One way to do this would be to internally normalize all path separators when reading metadata.

Are these security reasons why this would be a bad idea? What do you guys think?

**Comment by SantiagoTorres** [comment]
@trishankatdatadog let me see if I understand. you're suggesting to not normalize the paths upon writing, but rather normalize them upon reading the metadata? (probably upon verification)

I don't think there's anything wrong with that. Do you see any benefits for doing it though?

**Comment by trishankatdatadog** [comment]
@SantiagoTorres Sorry, I wasn't clear. Let's do both :) So let's normalize when writing, but _also_ when reading. I was just worried about being too unforgiving with the unexpected inputs (unless there were good security reasons, of course).

**Comment by SantiagoTorres** [comment]
We could do that, yes. I don't see anything against that. We'd have to make sure that both path separators are invalid characters in all platforms, which I think is true.

**Comment by SantiagoTorres** [comment]
@alyptik I dismissed your review since this PR was a complete rewrite in the end. Could you still review and ticketize anything I may have missed? Thanks!

File: tests/scripts/expr
Comment (lukpueh) [inline tests/scripts/expr:27]: I think a doc-copy-paste mistake happened here.

**Pull request merged**
